sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -225,7 +227,7 @@ public void updateAddresses(final URI serverKey) {
             } else {
                 forcedRefreshMetadata.signalPartitionAddressOnlyRefresh(partitionKeyRangeIdentity);
             }
-        } else if (request.forceCollectionRoutingMapRefresh) {
+        } else if (request.forceCollectionRoutingMapRefresh && partitionKeyRangeIdentity.getCollectionRid() != null) {
```

Comment:
maybe move the collectionRid check to the top?
if (partitionKeyRangeIdentity.getCollectionRid() != null) {
     //all the other logic

}

Reply:
Done.